### PR TITLE
Backport total_memory improvements from anaconda

### PR DESCRIPTION
**Pull request by VladimirSlavik**
The version used is a merger of the original blivet code and anaconda's counterpart from commit f22386ef. Note that the referenced commit has not been merged, and so isn't reachable in the repository itself. See below for relevant GitHub links. Note also that the function in anaconda has been retired in favor of using the blivet's one.

New:
- checks for bad input
- check for no data
- tests

See also:
- https://github.com/rhinstaller/anaconda/pull/4338
- https://github.com/rhinstaller/anaconda/commit/f22386efafe33df7f8182e65aaf99279904344c6

### CODE REVIEW

**Comment by StorageGhoul** [comment]
Can one of the admins verify this patch?

**Comment by vojtechtrefny** [comment]
Jenkins, ok to test.

**Review by vojtechtrefny — APPROVED** [review]
Looks good to me, just two small nitpicks.

**Comment by vojtechtrefny** [comment]
> What to focus on with review:
> 
>     * I don't know if the logging messages should be there.

I think the exceptions should be enough, we don't need to log it.

> 
>     * I believe the function could previously raise StopIteration, and now it's RuntimeError. However, it is never called such that the changed exceptions would matter. Still, it would be good to verify.

That should be OK.

Thanks for the fixed, definitely an improvement.

File: blivet/util.py [hunk 1]
```diff
@@ -305,23 +305,42 @@ def get_mount_device(mountpoint):
...
-    return mem
+    with open("/proc/meminfo", "r") as fobj:
+        for line in fobj:
+            if not line.startswith("MemTotal"):
+                # we are only interested in the MemTotal: line
+                continue
+
+            fields = line.split()
+            if len(fields) != 3:
+                log.error("unknown format for MemTotal line in /proc/meminfo: %s", line.rstrip())
```
[the remark this answers is not in the document]
Comment (VladimirSlavik) [reply blivet/util.py:325]: OK, let's do that.

**Comment by VladimirSlavik** [comment]
Updated per the above. I only wonder, should the tests be extended also to account for the "retries"?

**Review by vojtechtrefny — APPROVED** [review]

**Pull request merged**